src/structures/AwesomeFileSystem.cpp

```diff
@@ -28,26 +28,85 @@ void AwesomeFileSystem::load_superblock_into_memory() {
     fs_file << superblock.get_free_blocks().to_string();
     fs_file.seekg(0);
 }
+void AwesomeFileSystem::loop_for_write(int start, int end, string data, int address, int index = 0){
+    fs_file.seekg(address);
+    for(int i = start; i < end; i++){
+        fs_file.put(data[i + index]);
+    }
+};
+void AwesomeFileSystem::update_inode(Inode& inode, int size, int new_address){
+    inode.increase_blocks_amount();
+    inode.add_size_to_sizeof_file(size);
+    inode.update_storage_blocks(new_address);
+};
 
 void AwesomeFileSystem::write_to_file(string src_name, string data) { 
-    Inode inode = inode_map.get_inode(src_name).value();
-    std::vector<size_t> storage = inode.get_storage_blocks();
-    fs_file << data;
+    Inode inode = open_file(src_name);
+    int sizeof_file = inode.get_sizeof_file();
+    int blocks_amount = inode.get_blocks_amount();
+
+    int new_size = sizeof_file + data.size(); //new size for file
+    int available_memory = blocks_amount * BLOCK_SIZE; //for file on the current time
+    int difference = new_size - available_memory; // check who is bigger
+
+    if(difference <= 0){ //if we can write all data into last block
+        int free_memory_in_last_block = available_memory - sizeof_file;
+        int shift = BLOCK_SIZE - free_memory_in_last_block; // shift for address from start of the last block
+        size_t address = inode.get_last_block() + shift;
+        loop_for_write(0, data.size(), data, address);
+        fs_file.seekg(0);
+        //update info in inode
+        inode.add_size_to_sizeof_file(data.size());
+        //TODO: update time_t fields in inode
+    } else{
+        int extra_blocks = difference/BLOCK_SIZE; //the integer part of the number (extra blocks for data)
```

Comment:
You get the following logic: if we can cram the entire data into the last block, then we do it, if not, then we look for the next free block. But what if we have a situation that our last block is not full, but received data is too big for the rest memory in it? In this case, you will not use the rest of the last block, but will simply look for the next one, which is why you will lose a lot of memory.